src/Database/LSMTree/Internal/Vector.hs

```diff
@@ -26,3 +30,14 @@ noRetainedExtraMemory (PV.Vector off len ba) =
     off == 0 && len * sizeof == sizeofByteArray ba
    where
     sizeof = I# (sizeOfType# (Proxy @a))
+
+-- | /( O(n) /) Like 'V.map', but strict in the produced elements of type @b@.
+mapStrict :: forall a b. (a -> b) -> V.Vector a -> V.Vector b
+mapStrict f xs = V.create (VM.unsafeNew (V.length xs) >>= loop 0)
+  where
+    loop :: Int -> VM.MVector s b -> ST s (VM.MVector s b)
+    loop !i !ys
+      | i == V.length xs = pure ys
+      | otherwise        = do
+        VM.unsafeWrite ys i $! f (V.unsafeIndex xs i)
+        loop (i + 1) ys
```

Comment:
```suggestion
mapStrict f v = runST (V.mapM (\x -> pure $! f x) v)
```
I have benchmarked this against the above implementation. They get the same running times.
```
      Cr.env (pure $ V.fromList [1..1000 :: Int]) $ \v ->
        bgroup "vector" [
          bench "lazy map" (Cr.nf (V.map (+1)) v)
        , bench "loop map" (Cr.whnf (mapStrict (+1)) v)
        , bench "ST mapM"  (Cr.whnf (mapStrict' (+1)) v)
        ]
```